launcher/signature-discovery/client/client.go

```diff
@@ -0,0 +1,64 @@
+// Package client contains functionalities to discover container image signatures.
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/images"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+const signatureTagSuffix = "sig"
+
+// Client is a wrapper of containerd.Client to interact with signed image manifest.
+type Client struct {
+	cdClient      *containerd.Client
+	OriginalImage containerd.Image
+	RemoteOpts    []containerd.RemoteOpt
+}
+
+// New creates a new client.
+func New(cdClient *containerd.Client, originalImage containerd.Image, opts ...containerd.RemoteOpt) *Client {
+	c := &Client{
+		cdClient:      cdClient,
+		OriginalImage: originalImage,
+		RemoteOpts:    opts,
+	}
+	return c
+}
+
+// FetchSignedImageManifest fetches a signed image manifest using a tag-based discovery mechanism.
+func (c *Client) FetchSignedImageManifest(ctx context.Context, targetRepository string) (v1.Manifest, error) {
```

Comment:
Is this the function going to be called periodically?

Reply:
> Is this the function going to be called periodically?

Yes, this function will be ultimately called inside `refreshToken`. This change will be reflected on the following PR.